solidity/src/FastTransferGateway.sol

```diff
@@ -266,6 +266,8 @@ contract FastTransferGateway is Initializable, UUPSUpgradeable, OwnableUpgradeab
     /// @param repaymentAddress The address to repay the orders to
     /// @param orderIDs The IDs of the orders to settle
     function initiateSettlement(bytes32 repaymentAddress, bytes memory orderIDs) public payable {
```

Comment:
It would also be nice to change _settleOrder to tally the amount to send and modify order status in the same loop like in the cosmwasm side for symmetry and to prevent the duplicate orderID issue at message receive independently of the message send implementation. Will leave it up to your discretion.

Reply:
Yeah you're right, I updated this